lib/ruby_smb/client.rb

```diff
@@ -123,11 +171,17 @@ def initialize(dispatcher, smb1: true, smb2: true, username:, password:, domain:
       @username          = username.encode('utf-8') || ''.encode('utf-8')
       @max_buffer_size   = MAX_BUFFER_SIZE
 
+      negotiate_version_flag = 0x02000000
```

Comment:
The NEGOTIATE_VERSION flag is missing in rubyntlm library. I had to manually add it. I also submitted a PR to add it: https://github.com/WinRb/rubyntlm/pull/38